catalystwan/models/configuration/config_migration.py

```diff
@@ -167,6 +188,89 @@ def insert_parcel_type_from_headers(cls, values: Dict[str, Any]):
+            Union[
+                CentralizedPolicyInfo,
+                LocalizedPolicyInfo,
+                AnySecurityPolicyInfo,
+                AnyPolicyDefinitionInfo,
+                AnyPolicyListInfo,
+            ]
+        ] = None,
+    ):
+        self.failed_conversion_parcels.append(
+            FailedConversionParcel(
+                feature_template=feature_template,
+                policy=policy,
+                exception_message=exception_message,
+            )
+        )
+
+
+class ConfigGroupRaport(BaseModel):
+    model_config = ConfigDict(populate_by_name=True, extra="forbid")
+    name: str = Field(
+        serialization_alias="ConfigGroupName",
+        validation_alias="ConfigGroupName",
+        description="Name of the Config Group created from Device Template",
+    )
+    uuid: UUID = Field(
+        serialization_alias="ConfigGroupUuid",
+        validation_alias="ConfigGroupUuid",
+        description="UUID of the Config Group created from Device Template",
+    )
+    feature_profiles: List[FeatureProfileBuildRaport] = Field(
+        default=[],
+        serialization_alias="FeatureProfiles",
+        validation_alias="FeatureProfiles",
+        description="List of Feature Profiles created from Device Template and attached to the Config Group",
+    )
+
+
+class UX2ConfigPushRaport(BaseModel):
+    model_config = ConfigDict(populate_by_name=True, extra="forbid")
+    config_groups: List[ConfigGroupRaport] = Field(
+        default_factory=list, serialization_alias="ConfigGroups", validation_alias="ConfigGroups"
+    )
+    success_rate_message: str = Field(
+        default="", serialization_alias="SuccessRateMessage", validation_alias="SuccessRateMessage"
+    )
+    failed_push_parcels: List[FailedParcel] = Field(
+        default_factory=list, serialization_alias="FailedPushParcels", validation_alias="FailedPushParcels"
+    )
+    failed_converion_parcels: List[FailedConversionParcel] = Field(
+        default_factory=list, serialization_alias="FailedConversionParcels", validation_alias="FailedConversionParcels"
+    )
+    created_parcels: int = Field(default=0, serialization_alias="CreatedParcels", validation_alias="CreatedParcels")
+    failed_parcels: int = Field(default=0, serialization_alias="FailedParcels", validation_alias="FailedParcels")
+
+    def add_rapport(self, name: str, uuid: UUID, feature_profiles: List[FeatureProfileBuildRaport]) -> None:
+        self.config_groups.append(ConfigGroupRaport(name=name, uuid=uuid, feature_profiles=feature_profiles))
+
+    def set_push_success_rate(self):
+        for config_group in self.config_groups:
+            for feature_profile in config_group.feature_profiles:
+                self.created_parcels += len(feature_profile.created_parcels)
+                self.failed_parcels += len(feature_profile.failed_parcels)
+        all_parcels = self.created_parcels + self.failed_parcels
+        self.success_rate_message = (
```

Comment:
the member 'success_rate_message' is not used anywhere, is that right?
Also wondering, if wouldn't be better to not store this message in class member, but just return the current stats by 'get_success_rate_message' or sth similar

Reply:
This field will be send to the frontend